File: ClientApp/src/app/model/recipe.ts
import {Category} from "./category.enum";
import {Difficulty} from "./difficulty.enum";
import {RecipeImage} from "./recipe-image";

export class Recipe {
  id?: number;
  title?: string;
  isFavourite?: boolean;
  category: Category = Category.MAIN_DISHES;
  difficulty?: Difficulty = Difficulty.Hard;
  content?: string;
  shortDescription?: string;
  preparationTimeMinutes?: number;
  portions?: number;
  images: RecipeImage[] = [];
}

File: Models/Recipe.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace YouCanCook.Models;

public class Recipe
{
    [Key] public long Id { get; set; }

    [Required] public string? Title { get; set; }

    [Required] public string? Author { get; set; }

    public List<string> Ingredients { get; set; } = new();

    public string? Category { get; set; }
    
    public string? Difficulty { get; set; }
    
    public int? PreparationTimeMinutes { get; set; }
    public int? Portions { get; set; }
    
    public string? ShortDescription { get; set; }

    public string? Content { get; set; }

    public List<RecipeImage>? Images { get; set; }

    //TODO: call user info service and add cron to update display name  
    //[NotMapped]
    //public string AuthorName { get; set; }

    [NotMapped] public bool IsFavourite { get; set; }
}